cpp/src/io/json/nested_json_gpu.cu

```diff
@@ -593,8 +618,10 @@ struct PdaSymbolToSymbolGroupId {
     // The relative symbol group id of the current input symbol
     constexpr auto pda_sgid_lookup_size =
       static_cast<int32_t>(sizeof(tos_sg_to_pda_sgid) / sizeof(tos_sg_to_pda_sgid[0]));
+    auto const symbol_position =
+      symbol == delimiter ? static_cast<int32_t>('\n') : static_cast<int32_t>(symbol);
```

Comment:
so here we replace the delimiters with '\n'?

Reply:
The bugs come crawling out of the woodwork - thank you so much for pointing this out!
We don't replace it in the input. The intended behaviour is that `tos_sg_to_pda_sgid` mapping for delimiter and newline characters get swapped - when the delimiter character is encountered, we need to map it to `LINE_BREAK` in symbol group id (line 476), and the newline character gets mapped to `OTHER`. This is assuming that the delimiter character is not opening (closing) brace (bracket), quote, escape character, comma or colon.
With the above code, both newline and delimiter is mapped to `LINE_BREAK` which is incorrect. I've pushed the fix, along with a test that has both delimiter and newline characters in the input.